### PR TITLE
Added homework lesson21.

### PR DESCRIPTION
**Pull request by iBubbleGun**
Homework lesson21 added.
Please, check it.

### CODE REVIEW

File: src/main/java/edu/hillel/homework/lesson21/dao/DataBaseConnection.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    public static Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(URL, USERNAME, PASSWORD);
+    }
+
+    public static Connection getConnection(String db_url, String db_username, String db_password) throws SQLException {
+        return DriverManager.getConnection(db_url, db_username, db_password);
+    }
+
+    public static void close(Connection connection) throws SQLException {
```
Comment (kartius) [inline src/main/java/edu/hillel/homework/lesson21/dao/DataBaseConnection.java]: Please, implement AutoClosable interface by this class and override close method to close connection properly

File: src/main/java/edu/hillel/homework/lesson21/dao/HomeworkDao.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class HomeworkDao {
+
+    public void addHomework(@NotNull Homework homework) {
+        try (Connection connection = DataBaseConnection.getConnection();
```
Comment (kartius) [inline src/main/java/edu/hillel/homework/lesson21/dao/HomeworkDao.java]: Please, send connection via constructor, the HomeworkDao shouldn't know about DataBaseConnection, it should be incapsulated

File: src/main/java/edu/hillel/homework/lesson21/dao/LessonDao.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class LessonDao {
+
+    public void addLesson(@NotNull Lesson lesson) {
+        try (Connection connection = DataBaseConnection.getConnection();
```
Comment (kartius) [inline src/main/java/edu/hillel/homework/lesson21/dao/LessonDao.java]: The same here

File: src/main/java/edu/hillel/homework/lesson21/dao/DataBaseConnection.java (hunk 1 above)
Comment (iBubbleGun) [reply src/main/java/edu/hillel/homework/lesson21/dao/DataBaseConnection.java]: Well done.

File: src/main/java/edu/hillel/homework/lesson21/dao/HomeworkDao.java (hunk 2 above)
Comment (iBubbleGun) [reply src/main/java/edu/hillel/homework/lesson21/dao/HomeworkDao.java]: Well done.

File: src/main/java/edu/hillel/homework/lesson21/dao/LessonDao.java (hunk 3 above)
Comment (iBubbleGun) [reply src/main/java/edu/hillel/homework/lesson21/dao/LessonDao.java]: Well done.

**Pull request merged**